runDocker.ts

```diff
@@ -78,9 +79,34 @@ export async function removeContainer(containerId: string) {
   await docker.getContainer(containerId).remove()
 }
 
+async function pullImage(imageName: string) {
+  console.log(`Checking for and pulling ${imageName}`)
+  await new Promise((resolve, reject) =>
+    docker.pull(
+      imageName,
+      (pullError: Error, stream: NodeJS.ReadableStream) => {
+        docker.modem.followProgress(stream, onFinished)
+        function onFinished(err: Error | null, output: unknown[]) {
+          if (!err) {
+            console.log(output)
+            resolve(true)
+            return
+          }
+          reject(err)
+        }
+      }
+    )
+  )
```

Comment:
I think you can do this more succinctly with [`util.promisify`](https://nodejs.org/api/util.html#utilpromisifyoriginal):

```suggestion
  await util.promisify(docker.pull)(imageName)
```